acceptancetests/assess_caas_deploy_charms.py

```diff
@@ -12,213 +12,73 @@
----
-
-apiVersion: rbac.authorization.k8s.io/v1beta1
-kind: ClusterRoleBinding
-metadata:
-  name: hostpath-provisioner
-  namespace: kube-system
-subjects:
-  - kind: ServiceAccount
-    name: hostpath-provisioner
-    namespace: kube-system
-roleRef:
-  kind: ClusterRole
-  name: hostpath-provisioner
-  apiGroup: rbac.authorization.k8s.io
----
-
-apiVersion: rbac.authorization.k8s.io/v1beta1
-kind: Role
-metadata:
-  name: hostpath-provisioner
-  namespace: kube-system
-rules:
-  - apiGroups: [""]
-    resources: ["secrets"]
-    verbs: ["create", "get", "delete"]
----
-
-apiVersion: rbac.authorization.k8s.io/v1beta1
-kind: RoleBinding
-metadata:
-  name: hostpath-provisioner
-  namespace: kube-system
-roleRef:
-  apiGroup: rbac.authorization.k8s.io
-  kind: Role
-  name: hostpath-provisioner
-subjects:
-  - kind: ServiceAccount
-    name: hostpath-provisioner
----
-
-# -- Create a daemon set for web requests and send them to the nginx-ingress-controller
-apiVersion: extensions/v1beta1
-kind: DaemonSet
-metadata:
-  name: hostpath-provisioner
-  namespace: kube-system
-spec:
-  revisionHistoryLimit: 3
-  template:
-    metadata:
-      labels:
-        app: hostpath-provisioner
-    spec:
-      serviceAccountName: hostpath-provisioner
-      terminationGracePeriodSeconds: 0
-      containers:
-        - name: hostpath-provisioner
-          image: mazdermind/hostpath-provisioner:latest
-          imagePullPolicy: "IfNotPresent"
-          env:
-            - name: NODE_NAME
-              valueFrom:
-                fieldRef:
-                  fieldPath: spec.nodeName
-            - name: PV_DIR
-              value: /mnt/kubernetes
-          volumeMounts:
-            - name: pv-volume
-              mountPath: /mnt/kubernetes
-      volumes:
-        - name: pv-volume
-          hostPath:
-            path: /mnt/kubernetes
----
-
-# -- Create the standard storage class for running on-node hostpath storage
-apiVersion: storage.k8s.io/v1
-kind: StorageClass
-metadata:
-  # namespace: kube-system
-  name: {class_name}
-  annotations:
-    storageclass.beta.kubernetes.io/is-default-class: "true"
-  labels:
-    kubernetes.io/cluster-service: "true"
-    addonmanager.kubernetes.io/mode: EnsureExists
-provisioner: hostpath
----
-"""
 
+def check_app_healthy(url, timeout=300, success_hook=lambda: None, fail_hook=lambda: None):
+    if not callable(success_hook) or not callable(fail_hook):
+        raise RuntimeError("hooks are not callable")
 
-def check_app_healthy(url, timeout=300):
     status_code = None
-    for _ in until_timeout(timeout):
+    for remaining in until_timeout(timeout):
         try:
             r = requests.get(url)
             if r.ok and r.status_code < 400:
-                return
+                return success_hook()
             status_code = r.status_code
         except IOError as e:
             log.error(e)
-        sleep(3)
+        finally:
+            sleep(3)
+            if remaining % 60 == 0:
```

Comment:
perhaps 60 is too long, maybe 15? Or 30?

Reply:
changed to 30s